### PR TITLE
Add class pr curves

### PR DESCRIPTION
**Pull request by roshankern**
This is ready for review! This PR creates precision-recall curves for every phenotypic class using the final and shuffled baseline models on the training and testing data.

### CODE REVIEW

File: 3.evaluate_model/evaluate_model.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-# Convert notebook to python file and execute
+# Convert notebooks to python file and execute
 jupyter nbconvert --to python \
         --FilesWriter.build_directory=scripts/nbconverted \
-        --execute evaluate_model.ipynb
+        --execute *.ipynb
```
Comment (gwaybio) [inline 3.evaluate_model/evaluate_model.sh]: it is dangerous to have this wildcard here 😨 (notebooks run that you don't necessarily intend)

If there are multiple notebooks to execute, please list them explicitly. Doing so also ensures correct order of operations, if necessary.

File: 3.evaluate_model/scripts/nbconverted/class_PR_curves.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+# ### PR Curves for combinations of models and data subsets
+
+# In[3]:
+
+
+fig, PR_data  = class_PR_curves(train_data, log_reg_model)
+fig.suptitle("Precision Recall Curves for Final Model on Training Data")
+fig.savefig(f"{save_dir}/final_model_train_data.png")
+PR_data.to_csv(f"{save_dir}/final_model_train_data.tsv", sep="\t")
```
Comment (gwaybio) [inline 3.evaluate_model/scripts/nbconverted/class_PR_curves.py:52]: Can you please add a sneak peek into what `PR_data` looks like in this notebook?

This will suffice:

```python
print(PR_data.shape)
PR_data.head()
```

File: utils/evaluate_utils.py [hunk 3]
```diff
@@ -13,12 +13,83 @@
...
+    -------
+    pd.DataFrame
+        dataframe with precision/recall data for each class in tidy long format
+
+    matplotlib.figure.Figure
+        figure with compilation of all class PR curves
+    """
+
+    phenotypic_classes = log_reg_model.classes_
+    X, y = get_X_y_data(single_cell_data)
```
Comment (gwaybio) [inline utils/evaluate_utils.py:47]: If this takes a while to load, you can consider loading outside the function (in the notebook) and take X, y as input to the function. If it is quick, then no worries.

**Review by gwaybio — APPROVED** [review]
LGTM! Thanks for such a nice, bite-sized PR. Happy to take another look prior to merging if you'd like

File: utils/evaluate_utils.py [hunk 4]
```diff
@@ -13,12 +13,83 @@
...
+from sklearn.preprocessing import label_binarize
+
 # set numpy seed to make random operations reproduceable
 np.random.seed(0)
 
-def evaluate_model_cm(
-    log_reg_model: LogisticRegression, dataset: pd.DataFrame
-):
+
+def class_PR_curves(single_cell_data: pd.DataFrame, log_reg_model):
```
Comment (d33bs) [inline utils/evaluate_utils.py]: Consider adding the `log_reg_model` and return type hints here.

File: utils/evaluate_utils.py [hunk 5]
```diff
@@ -13,12 +13,83 @@
 
 from train_utils import get_X_y_data
 
+from sklearn.metrics import precision_recall_curve
+from sklearn.preprocessing import label_binarize
```
Comment (d33bs) [inline utils/evaluate_utils.py]: It might make sense to combine the `sklearn.metrics` import here with the earlier line in this file and move the `sklearn.preprocessing` line to be grouped with those earlier lines.

File: utils/evaluate_utils.py [hunk 6]
```diff
@@ -13,12 +13,83 @@
...
+
+    # only label outer plots
+    for ax in axs.flat:
+        ax.label_outer()
+
+    PR_data = pd.concat(PR_data, axis=0).reset_index(drop=True)
+    return fig, PR_data
+
+
+def evaluate_model_cm(log_reg_model: LogisticRegression, dataset: pd.DataFrame):
```
Comment (d33bs) [inline utils/evaluate_utils.py]: Consider adding a return type hint for this function.

File: utils/evaluate_utils.py [hunk 7]
```diff
@@ -13,12 +13,83 @@
...
+        precision, recall, _ = precision_recall_curve(y_binarized[:, i], y_score[:, i])
+        class_tidy_data = pd.DataFrame(
+            {
+                "Phenotypic_Class": phenotypic_classes[i],
+                "PR_Threshold": PR_threshold,
+                "Precision": precision,
+                "Recall": recall,
+            }
+        )
+        PR_data.append(class_tidy_data)
```
Comment (d33bs) [inline utils/evaluate_utils.py]: If the variable `class_tidy_data` doesn't get used again within this loop after being appended to `PR_data`, think about adding the `pd.DataFrame` directly to your append call (without a variable assigned). The difference here might be minor, and is completely optional, but could save a line or two of code and conserves the use of variables within the loop.

**Review by d33bs — COMMENTED** [review]
Nice work! I left a few minor comments but things mostly looked good to me. Please don't hesitate to let me know if you have any questions.

File: utils/evaluate_utils.py (hunk 3 above)
Comment (roshankern) [reply utils/evaluate_utils.py:47]: This is quick :)
